From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Max Lee <yowching.lee@gmail.com>
Date: Tue, 9 Jun 2026 02:30:00 +0000
Subject: [PATCH] use C-compatible lisp function typedef

---
 src/lispbibl.d | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lispbibl.d b/src/lispbibl.d
index 111111111..222222222 100644
--- a/src/lispbibl.d
+++ b/src/lispbibl.d
@@ -9324,11 +9324,11 @@ typedef gcv_object_t symbol_;
   /* To pass a type of the value Values: return_Values(...); */
   #define return_Values  return_void
   /* A Lisp-function is a pointer to a C-function without returned value. */
-  typedef Values (*lisp_function_t)(...);
+  typedef Values (*lisp_function_t)();
 /* If this is changed, every call of a C-function with the result type
  'Values' (especially 'funcall', 'apply', 'eval') is to be checked. */
 %% puts("typedef void Values;"); /* emit_typedef useless: no sizeof(void) */
-%% emit_typedef_f("Values (*%s)(...)","lisp_function_t");
+%% emit_typedef_f("Values (*%s)()","lisp_function_t");
 
 /* FSUBRs
  As C-functions: of type fsubr_function_t (no arguments, no value): */
-- 
2.47.1
