From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Max Lee <yowching.lee@gmail.com>
Date: Tue, 9 Jun 2026 05:20:00 +0000
Subject: [PATCH] disable impnotes network check

---
 src/makemake.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/makemake.in b/src/makemake.in
index 333333333..444444444 100644
--- a/src/makemake.in
+++ b/src/makemake.in
@@ -3679,5 +3679,5 @@ if [ "$TSYS" != win32msvc ] ; then
   MYIMPROOT=http://www.gnu.org/software/clisp/impnotes/
   MYTESTURL=${MYIMPROOT}id-href.map
-  if test -d "${SRCTOPDIR_}.git" \
+  if false \
      && { if wget --version >/dev/null 2>/dev/null ; then \
             wget --execute netrc=off --tries=1 --timeout=10 -q ${MYTESTURL}; \
           elif lynx --version >/dev/null 2>/dev/null ; then \
-- 
2.47.1
