From b1ef32038b988a32e563a2be099422219ee2cda0 Mon Sep 17 00:00:00 2001
From: Max Lee <yowching.lee@gmail.com>
Date: Fri, 10 Jul 2026 12:52:40 +0800
Subject: [PATCH] link iconv library if enabled

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index d93080d4..1e4adaa4 100755
--- a/configure
+++ b/configure
@@ -907,8 +907,7 @@ if [ "$addflags" = "yes" ] ; then

   case "${flagsm}" in
     *le|*gnu)
-        LIBS="${LIBS} -lm -ldl ${ncursesLib} -lrt"
+        LIBS="${LIBS} -lm -ldl ${iconvLib} ${ncursesLib} -lrt"
         ;;
     *fb|*ob|*hk)
         LIBS="${LIBS} ${iconvLib} -lm ${ncursesLib}"
--
2.30.2
